NoSuchMethodError `setPixels()`

### Steps to reproduce

- Download J-Novel 1.4.1 extension
- Open a free chapter and try to load the images

### Expected behavior

Images should load.

### Actual behavior

```
2026-05-03 01:03:59.902 [DefaultDispatcher-worker-14] ERROR suwayomi.tachidesk.server.JavalinSetup - IOException while handling the request
java.io.IOException: canceled due to java.lang.NoSuchMethodError: 'void android.graphics.Bitmap.setPixels(int[], int, int, int, int, int, int)'
	at eu.kanade.tachiyomi.source.online.HttpSource.getImage$suspendImpl(HttpSource.kt:374)
	at eu.kanade.tachiyomi.source.online.HttpSource.getImage(HttpSource.kt)
	at suwayomi.tachidesk.manga.impl.Page$getPageImage$6.invokeSuspend(Page.kt:138)
	at suwayomi.tachidesk.manga.impl.Page$getPageImage$6.invoke(Page.kt)
	at suwayomi.tachidesk.manga.impl.Page$getPageImage$6.invoke(Page.kt)
	at suwayomi.tachidesk.manga.impl.util.storage.ImageResponse.getImageResponse(ImageResponse.kt:69)
	at suwayomi.tachidesk.manga.impl.Page.getPageImage(Page.kt:137)
	at suwayomi.tachidesk.manga.impl.Page.getPageImage$default(Page.kt:57)
	at suwayomi.tachidesk.manga.impl.Page.getPageImageServe(Page.kt:149)
	at suwayomi.tachidesk.manga.controller.MangaController$pageRetrieve$2$1$1.invokeSuspend(MangaController.kt:478)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:34)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:124)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:89)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:586)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:820)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:717)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:704)
Caused by: java.io.IOException: canceled due to java.lang.NoSuchMethodError: 'void android.graphics.Bitmap.setPixels(int[], int, int, int, int, int, int)'
	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:587)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.NoSuchMethodError: 'void android.graphics.Bitmap.setPixels(int[], int, int, int, int, int, int)'
	at eu.kanade.tachiyomi.extension.en.jnovel.XebpDecoder.compositePatch(Unknown Source)
	at eu.kanade.tachiyomi.extension.en.jnovel.XebpDecoder.decrypt(Unknown Source)
	at eu.kanade.tachiyomi.extension.en.jnovel.ImageInterceptor.intercept(Unknown Source)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:126)
	at eu.kanade.tachiyomi.network.interceptor.CloudflareInterceptor.intercept(CloudflareInterceptor.kt:44)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:126)
	at eu.kanade.tachiyomi.network.interceptor.UserAgentInterceptor.intercept(UserAgentInterceptor.kt:21)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:126)
	at eu.kanade.tachiyomi.network.interceptor.UncaughtExceptionInterceptor.intercept(UncaughtExceptionInterceptor.kt:18)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:126)
	at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:226)
	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:574)
	... 3 common frames omitted
2026-05-03 01:03:59.917 [OkHttp Dispatcher] ERROR suwayomi.tachidesk.server.ServerSetup - unhandled exception
java.lang.NoSuchMethodError: 'void android.graphics.Bitmap.setPixels(int[], int, int, int, int, int, int)'
	at eu.kanade.tachiyomi.extension.en.jnovel.XebpDecoder.compositePatch(Unknown Source)
	at eu.kanade.tachiyomi.extension.en.jnovel.XebpDecoder.decrypt(Unknown Source)
	at eu.kanade.tachiyomi.extension.en.jnovel.ImageInterceptor.intercept(Unknown Source)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:126)
	at eu.kanade.tachiyomi.network.interceptor.CloudflareInterceptor.intercept(CloudflareInterceptor.kt:44)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:126)
	at eu.kanade.tachiyomi.network.interceptor.UserAgentInterceptor.intercept(UserAgentInterceptor.kt:21)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:126)
	at eu.kanade.tachiyomi.network.interceptor.UncaughtExceptionInterceptor.intercept(UncaughtExceptionInterceptor.kt:18)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:126)
	at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:226)
	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:574)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)

```

### Suwayomi-Server version

v2.1.2092

### Used database

H2

### Server operating system

Windows 11 Pro 25H2

### Server Desktop Environment

_No response_

### Server JVM version

openjdk 21.0.5 2024-10-15 LTS

### Used client name

WebUI

### Client version

r2998

### Used web browser

Chrome 147.0.7727.138

### Client operating system

Windows 11 Pro 25H2

### Other details

Add `Bitmap.setPixels()` functionality

### Acknowledgements

- [x] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [x] I have written a short but informative title (ideally less than ~100 characters).
- [x] I have tried the troubleshooting guide described in [README.md](https://github.com/Suwayomi/Suwayomi-Server?tab=readme-ov-file#troubleshooting-and-support)
- [x] I have updated to the **[latest version](https://github.com/suwayomi/suwayomi-server/releases/latest)**.
- [x] I have filled out all of the requested information in this form, including specific version numbers.
- [x] I understand that **Suwayomi does not have or fix any extensions**, and I **will not receive help** for any issues related to sources or extensions.

--- a/AndroidCompat/src/main/java/android/graphics/Bitmap.java
+++ b/AndroidCompat/src/main/java/android/graphics/Bitmap.java
@@ -345,6 +345,57 @@ public final class Bitmap {
         return image.getRGB(x, y);
     }
 
+    /**
+     * <p>Write the specified {@link Color} into the bitmap (assuming it is
+     * mutable) at the x,y coordinate. The color must be a
+     * non-premultiplied ARGB value in the {@link ColorSpace.Named#SRGB sRGB}
+     * color space.</p>
+     *
+     * @param x     The x coordinate of the pixel to replace (0...width-1)
+     * @param y     The y coordinate of the pixel to replace (0...height-1)
+     * @param color The ARGB color to write into the bitmap
+     *
+     * @throws IllegalStateException if the bitmap is not mutable
+     * @throws IllegalArgumentException if x, y are outside of the bitmap's
+     *         bounds.
+     */
+    public void setPixel(int x, int y, @ColorInt int color) {
+        checkPixelAccess(x, y);
+        image.setRGB(x, y, color);
+    }
+
+    /**
+     * <p>Replace pixels in the bitmap with the colors in the array. Each element
+     * in the array is a packed int representing a non-premultiplied ARGB
+     * {@link Color} in the {@link ColorSpace.Named#SRGB sRGB} color space.</p>
+     *
+     * @param pixels   The colors to write to the bitmap
+     * @param offset   The index of the first color to read from pixels[]
+     * @param stride   The number of colors in pixels[] to skip between rows.
+     *                 Normally this value will be the same as the width of
+     *                 the bitmap, but it can be larger (or negative).
+     * @param x        The x coordinate of the first pixel to write to in
+     *                 the bitmap.
+     * @param y        The y coordinate of the first pixel to write to in
+     *                 the bitmap.
+     * @param width    The number of colors to copy from pixels[] per row
+     * @param height   The number of rows to write to the bitmap
+     *
+     * @throws IllegalStateException if the bitmap is not mutable
+     * @throws IllegalArgumentException if x, y, width, height are outside of
+     *         the bitmap's bounds.
+     * @throws ArrayIndexOutOfBoundsException if the pixels array is too small
+     *         to receive the specified number of pixels.
+     */
+    public void setPixels(@NonNull @ColorInt int[] pixels, int offset, int stride,
+            int x, int y, int width, int height) {
+        if (width == 0 || height == 0) {
+            return; // nothing to do
+        }
+        checkPixelsAccess(x, y, width, height, offset, stride, pixels);
+        image.setRGB(x, y, width, height, pixels, offset, stride);
+    }
+
     public void eraseColor(int c) {
         java.awt.Color color = Color.valueOf(c).toJavaColor();
         Graphics2D graphics = image.createGraphics();

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -12,7 +12,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/).
 - (**Extension/Android**) Add Main dispatcher implementation
 - (**Extension/Android**) Add LruCache implementation
 - (**Extension/Android**) Support basic BitmapFactory options
-- (**Extension/Android**) Support Bitmap pixel-based access
+- (**Extension/Android**) Support Bitmap pixel-based access and modification
 - (**Extension/Android**) Add Rect.set functionality
 - (**OPDS**) Add reading progress synchronization for KOReader
 - (**WebView**) Support copy & paste

